com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/SetBreakpointsRequestHandler.java

```diff
@@ -170,45 +173,58 @@ private void registerBreakpointHandler(IDebugAdapterContext context) {
                     }
 
                     // find the breakpoint related to this breakpoint event
-                    IBreakpoint conditionalBP = Arrays.asList(manager.getBreakpoints()).stream().filter(bp -> StringUtils.isNotBlank(bp.getCondition())
```

Comment:
Please improve the logic here. The method is lengthy.

Reply:
fixed.